### PR TITLE
Phpstan fixes

### PR DESCRIPTION
**Pull request by erikn69**
Make PHPStan happy
On [phpstan:v1.10.12](https://github.com/phpstan/phpstan/releases/tag/1.10.12), there is new validations

https://github.com/spatie/laravel-permission/actions/runs/4681036769/jobs/8293124388

### CODE REVIEW

File: src/PermissionRegistrar.php [hunk 1]
```diff
@@ -25,7 +25,7 @@ class PermissionRegistrar
     /** @var string */
     protected $roleClass;
 
-    /** @var Collection|null */
+    /** @var array|null */
```
Comment (drbyte) [inline src/PermissionRegistrar.php:28]: Um ... but it's actually still a collection internally ...

Comment (erikn69) [reply src/PermissionRegistrar.php:28]: You are right, I see that the problem starts with the new version of phpstan

Comment (erikn69) [reply src/PermissionRegistrar.php]: Fixed

**Comment by szepeviktor** [comment]
> there is new validations

Which validation is it?

Without the var tag PHPStan does not see the contract.

```
Method Spatie\Permission\Models\Permission::getPermission() should return Spatie\Permission\Contracts\Permission|null but returns Illuminate\Database\Eloquent\Model|null.
```

**Pull request merged**